tests/Resources/util.addontest.js

```diff
@@ -0,0 +1,1210 @@
+			function BaseClass() {
+			}
+
+			should.throws(() => util.inherits(null, BaseClass),
+				TypeError
+			);
+		});
+
+		it('throws TypeError if super constructor has no prototype', () => {
+			const BaseClass = Object.create(null, {});
+
+			function MyStream() {
+				BaseClass.call(this);
+			}
+
+			should.throws(() => util.inherits(MyStream, BaseClass),
+				TypeError
+			);
+		});
+	});
+
+	describe('#promisify()', () => {
+		it('is a function', () => {
+			util.promisify.should.be.a.Function;
+		});
+
+		it('wraps callback function to return promise with resolve', (finished) => {
+			function callbackOriginal(argOne, argTwo, next) {
+				next(argOne, argTwo);
+			}
+			const promisified = util.promisify(callbackOriginal);
+			const result = promisified(null, 123);
+			should(result instanceof Promise).eql(true);
+			result.then(value => { // eslint-disable-line promise/always-return
+				should(value).eql(123);
+				finished();
+			}).catch(err => finished(err));
+		});
+
+		it('wraps callback function to return promise with rejection', (finished) => {
+			function callbackOriginal(argOne, argTwo, next) {
+				next(argOne, argTwo);
+			}
+			const promisified = util.promisify(callbackOriginal);
+			const result = promisified(new Error('example'), 123);
+			should(result instanceof Promise).eql(true);
+			result.then(value => { // eslint-disable-line promise/always-return
+				should(value).eql(123);
+				finished(new Error('Expected promise to get rejected!'));
+			}).catch(err => {
+				err.message.should.eql('example');
+				finished();
+			});
+		});
+
+		it('throws TypeError if original argument is not a function', () => {
+			should.throws(() => util.promisify({}),
+				TypeError
+			);
+		});
+	});
+
+	describe('#callbackify()', () => {
+		it('is a function', () => {
+			util.callbackify.should.be.a.Function;
+		});
+
+		it('wraps function returning Promise to return function accepting callback (with success)', (finished) => {
+			function original(argOne) {
+				return Promise.resolve(argOne);
+			}
+			const callbackified = util.callbackify(original);
+			callbackified(23, (err, result) => {
+				try {
+					should(err).not.be.ok;
+					should(result).eql(23);
+					finished();
+				} catch (e) {
+					finished(e);
+				}
+			});
+		});
+
+		it('wraps function returning Promise to return function accepting callback (with error)', (finished) => {
+			function original(argOne) {
+				return Promise.reject(argOne);
+			}
+			const callbackified = util.callbackify(original);
+			callbackified(new Error('expected this'), (err, result) => {
+				try {
+					should(err).be.ok;
+					should(result).not.be.ok;
+					finished();
+				} catch (e) {
+					finished(e);
+				}
+			});
+		});
+
+		it('handles special case of falsy rejection', (finished) => {
+			function original() {
+				return Promise.reject(null);
+			}
+			const callbackified = util.callbackify(original);
+			callbackified((err, result) => {
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/u...
  
  
  DangerID: danger-id-default;
  File: tests/Resources/util.addontest.js;
  Line: 881;
-->

- :warning: tests/Resources/util.addontest.js line 881 – 'result' is defined but never used. (no-unused-vars)